practices/managing-static-assets.md

```diff
@@ -1,16 +1,13 @@
+We host on our own infrastructure these kinds of static resources, instead of including them on our pages from an third-party CDN. This applies especially to JavaScript files or libraries, and font files.
 
-For this kind of static resources it is our choice to **self-host them on our servers** instead of including them on our pages from an external CDN.
+Self-hosting static assets has several advantages over relying on third parties for their delivery:
 
-## Benefits of self-hosting static assets
-
-* **Resilience.** As long as our site is up, the resource will be up. We don't depend on other servers or additional CDNs and we won't be affected by their outages, or closure of the service.
-* **Consistency.** Static assets are cached and served in exactly the same way that the reset of our resources are, and they use the same CDN. As a result, timings and user experience are more consistent all around the world.
-* **Security.** reduces our front-end [attack surface](https://en.wikipedia.org/wiki/Attack_surface).
-* **Performance.** Self-hosting can save us the cost of the DNS request, TCP connection and TLS negotiation, which in slow connections can easily be between 1000ms and 2000ms. This can also be mitigated with `rel=preconnect`.
-* **Filtering.** Some CDNs are filtered by the big firewall in China. This can slow down the loading of the page, or even make them break. Self-hosting static assets prevents this.
-* **Security.** Self-hosting the asset makes it easier for us to use [Subresource Integrity (SRI)](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) in the future.
+* **Resilience.** As long as our site is up, the static resource will be up too. We don't depend on third party servers or additional CDNs besides our own, and we won't be affected by their outages, or closure of the service.
+* **Consistency.** Static assets are cached and served like the rest of our resources are, and they use the same CDN. As a result, resource timings and user experience are more consistent all around the world.
+* **Security.** reduces our front-end [attack surface](https://en.wikipedia.org/wiki/Attack_surface) as the resources will enjoy the same level of security as the rest of first party content, and we don't depend on third parties being secure. Self-hosting the asset makes it easier for us to use [Subresource Integrity (SRI)](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) for an additional layer of security.
+* **Performance.** Self-hosting can save the users the cost of the DNS request, TCP connection and TLS negotiation for each self-hosted request, which in slow connections can make a big difference. This can also be mitigated with [`rel=dns-prefetch`](https://developer.mozilla.org/en-US/docs/Learn/Performance/dns-prefetch) and [`rel=preconnect`](https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preconnect), but self-hosting the assets eliminates that penalty completely.
```

Comment:
"Self-hosting can save the users the cost of the DNS request, TCP connection and TLS negotiation for each self-hosted request, which in slow connections can make a big difference."

Should this say "Self-hosting can save the users the cost of the DNS request, TCP connection and TLS negotiation for each **CDN-hosted** request, which in slow connections can make a big difference."

The cost of those requests is on the CDN, and we save those cost _by_ self-hosting?

Reply:
Does it not only save DNS, not TCP & TLS?

Reply:
I guess that will depend if it's a different subdomain and the cert covers all the (sub)domains, or a completely separate domain, but I don't think it makes sense to complicate the explanation so much.